api/src/main/java/net/kyori/adventure/text/Component.java

```diff
@@ -159,21 +160,43 @@
    * @param components the components
    * @return a text component
    * @since 4.0.0
+   * @deprecated for removal since 4.9.0, use {@link #join(JoinConfiguration, Iterable)} with a {@link JoinConfiguration} instead.
    */
+  @Deprecated
   @Contract(value = "_, _ -> new", pure = true)
   static @NotNull TextComponent join(final @NotNull ComponentLike separator, final Iterable<? extends ComponentLike> components) {
-    final Iterator<? extends ComponentLike> it = components.iterator();
-    if (!it.hasNext()) {
-      return Component.empty();
-    }
-    final TextComponent.Builder builder = text();
-    while (it.hasNext()) {
-      builder.append(it.next());
-      if (it.hasNext()) {
-        builder.append(separator);
-      }
-    }
-    return builder.build();
+    final Component component = join(JoinConfiguration.separator(separator), components);
+
+    if (component instanceof TextComponent) return (TextComponent) component;
+    return Component.text().append(component).build();
+  }
+
+  /**
+   * Joins {@code components} using the configuration in {@code config}.
+   *
+   * @param config the join configuration
+   * @param components the components
+   * @return a text component
+   * @see JoinConfiguration
```

Comment:
add @see for #noSeparators, #separator, and #separators here, replacing the one for `JoinConfiguration`.